src/chocolatey.resources/helpers/functions/Get-Installed.ps1

```diff
@@ -0,0 +1,48 @@
+﻿## Get-Installed
+##############################################################################################################
+## Returns a list of installed applications.
+## Registry based replacement for "Get-WmiObject -Class Win32_Product".
+##############################################################################################################
+
+function Get-Installed {
+param(
+  [switch] $GWMICompatible = $false # You may use '-GWMICompatible' switch
+)
+
+  $items = @()
+
+  if(Test-Path "HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall") {
+    $items += Get-ChildItem "HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall"
+  }
+  if(Test-Path "HKLM:SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall") {
+    $items += Get-ChildItem "HKLM:SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall"
+  }
+
+  $productList = @()
+
+  foreach ($app in $items)
+  {
+    $propitems = $app | Foreach-Object {Get-ItemProperty $_.PsPath}
+    $w32item = New-Object System.Object
+    $w32item | Add-Member -type NoteProperty -Name "IdentifyingNumber" -Value $app.PsChildname
+    $w32item | Add-Member -type NoteProperty -Name "Name"    -Value $propitems.DisplayName
+    $w32item | Add-Member -type NoteProperty -Name "Vendor"  -Value $propitems.Publisher
+    $w32item | Add-Member -type NoteProperty -Name "Version" -Value $propitems.DisplayVersion

```

Comment:
A prototype, and as opt-in. Exactly what I had stated before. Allow folks to opt into newer behavior that may not fully work as expected.